data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/oidc/OIDCDiscoveryConfig.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2018 Knative Authors (knative-dev@googlegroups.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.knative.eventing.kafka.broker.core.oidc;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.net.PemTrustOptions;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import org.jose4j.jwk.JsonWebKeySet;
+import org.jose4j.keys.resolvers.JwksVerificationKeyResolver;
+import org.jose4j.lang.JoseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class OIDCDiscoveryConfig {
+
+    private static final Logger logger = LoggerFactory.getLogger(TokenVerifier.class);
+
+    private String issuer;
+
+    private JwksVerificationKeyResolver jwksVerificationKeyResolver;
+
+    private OIDCDiscoveryConfig() {}
+
+    public String getIssuer() {
+        return issuer;
+    }
```

Comment:
Can we make this thread safe with an atomic reference ?

If the caller calls `getIssuer` while the pipeline created by build is running we have a data race

Reply:
Same for jwksVerificationKeyResolver

Reply:
Not sure if I understand you correctly, but `build` returns a future, which resolves with the OIDCDiscoveryConfig object. So not sure how to call `getIssuer()` before.

Reply:
Ah that's true, thanks!